go/mysql/flavor_mysqlgr.go

```diff
@@ -0,0 +1,222 @@
+			res.SecondsBehindMaster = math.MaxUint32
+		}
+		return nil
+	})
+	if err != nil {
+		return ReplicationStatus{}, err
+	}
+	// if chanel is not set, it means the state is not ONLINE or RECOVERING
+	// return partial result early
+	if chanel == "" {
+		return res, nil
+	}
+
+	// Populate IOThreadRunning from replication_connection_status
+	query = fmt.Sprintf(`SELECT SERVICE_STATE
+		FROM performance_schema.replication_connection_status
+		WHERE CHANNEL_NAME='%s'`, chanel)
+	var ioThreadRunning bool
+	err = fetchStatusForGroupReplication(c, query, func(values []sqltypes.Value) error {
+		ioThreadRunning = values[0].ToString() == "ON"
+		return nil
+	})
+	if err != nil {
+		return ReplicationStatus{}, err
+	}
+	res.IOThreadRunning = ioThreadRunning
+	// Populate SQLThreadRunning from replication_connection_status
+	var sqlThreadRunning bool
+	query = fmt.Sprintf(`SELECT SERVICE_STATE
+		FROM performance_schema.replication_applier_status_by_coordinator
+		WHERE CHANNEL_NAME='%s'`, chanel)
+	err = fetchStatusForGroupReplication(c, query, func(values []sqltypes.Value) error {
+		sqlThreadRunning = values[0].ToString() == "ON"
+		return nil
+	})
+	if err != nil {
+		return ReplicationStatus{}, err
+	}
+	res.SQLThreadRunning = sqlThreadRunning
+
+	// Collect lag information
+	// we use the difference between the last processed transaction's commit time
+	// and the end buffer time as the proxy to the lag
+	query = fmt.Sprintf(`SELECT
+		TIMESTAMPDIFF(SECOND, LAST_PROCESSED_TRANSACTION_ORIGINAL_COMMIT_TIMESTAMP, LAST_PROCESSED_TRANSACTION_END_BUFFER_TIMESTAMP)
+	FROM
+		performance_schema.replication_applier_status_by_coordinator
+	WHERE
+		CHANNEL_NAME='%s'`, chanel)
+	err = fetchStatusForGroupReplication(c, query, func(values []sqltypes.Value) error {
+		parseReplicationApplierLag(&res, values)
+		return nil
+	})
+	if err != nil {
+		return ReplicationStatus{}, err
+	}
+	return res, nil
+}
+
+func parsePrimaryGroupMember(res *ReplicationStatus, row []sqltypes.Value) {
+	res.MasterHost = row[0].ToString() /* MEMBER_HOST */
+	memberPort, _ := row[1].ToInt64()  /* MEMBER_PORT */
+	res.MasterPort = int(memberPort)
+}
+
+func parseReplicationApplierLag(res *ReplicationStatus, row []sqltypes.Value) {
+	lagSec, err := row[0].ToInt64()
+	// if the error is not nil, SecondsBehindMaster will remain to be MaxUint32
+	if err == nil {
+		// Only set where there is no error
+		// The value can be NULL when there is no replication applied yet
+		res.SecondsBehindMaster = uint(lagSec)
+	}
+}
+
+func fetchStatusForGroupReplication(c *Conn, query string, onResult func([]sqltypes.Value) error) error {
+	qr, err := c.ExecuteFetch(query, 100, true /* wantfields */)
+	if err != nil {
+		return err
+	}
+	// if group replication related query returns 0 rows, it means the group replication is not set up
+	if len(qr.Rows) == 0 {
+		return ErrNoGroupStatus
+	}
+	if len(qr.Rows) > 1 {
+		return fmt.Errorf("unexpected results for %v: %v", query, qr.Rows)
```

Comment:
we try to use `vterrors.Errors` instead of `fmt`

Reply:
changed to `vterrors.Errorf`